tests/e2e/default/sms/rapidpro.wdio-spec.js

```diff
@@ -0,0 +1,593 @@
+
+      const headers = broadcastsEndpointRequests.map(item => item[1]);
+      headers.forEach(header => expect(header.authorization).to.equal(`Token ${OUTGOING_KEY}`));
+    });
+
+    it('should set correct states from broadcast api', async () => {
+      await setOutgoingKey();
+
+      const statuses = ['queued', '', 'queued', 'sent', 'failed'];
+      broadcastsResult = (req, res) => {
+        const idx = req.body.text.replace('message', '');
+        res.json({
+          id: `broadcast${idx}`,
+          status: statuses[idx - 1],
+        });
+      };
+
+      const { rev } = await utils.saveDoc(pregnancyReportWithTasks);
+      const revNumber = parseInt(rev);
+      await browser.waitUntil(() => broadcastsEndpointRequests.length === 4, 1200);
+      await utils.waitForDocRev([{ id: pregnancyReportWithTasks._id, rev: revNumber + 1 }]);
+
+      const report = await utils.getDoc(pregnancyReportWithTasks._id);
+      verifyGatewayRefAndState(report.tasks[0], 'broadcast1', 'received-by-gateway');
+      verifyGatewayRefAndState(report.tasks[1], undefined, 'sent');
+      verifyGatewayRefAndState(report.scheduled_tasks[0], 'broadcast3', 'received-by-gateway');
+      verifyGatewayRefAndState(report.scheduled_tasks[1], 'broadcast4', 'sent');
+      verifyGatewayRefAndState(report.scheduled_tasks[2], 'broadcast5', 'failed');
+    });
+
+    it('should update the states correctly from messages api', async () => {
+      const genReport = (id, index) => genericReportFactory.reportWithTasks().build({
+        _id: id,
+        tasks: [{
+          messages: [{ to: `phone${index}`, message: `message${index}`, uuid: `uuid${index}` }],
+          gateway_ref: `gateway_ref_${id}`,
+          state: 'received-by-gateway',
+        }]
+      });
+
+      const docs = [
+        genReport('received_to_queued', 1),
+        genReport('received_to_wired', 2),
+        genReport('received_to_sent', 3),
+        genReport('received_to_resent', 4),
+        genReport('received_to_delivered', 5),
+        genReport('received_to_errored', 6),
+        genReport('received_to_failed', 7),
+      ];
+
+      messagesResult = (req, res) => {
+        const id = req.query.broadcast.replace('gateway_ref_', '');
+        const nextStatus = id.replace('received_to_', '');
+        res.json({ results: [{ status: nextStatus }] });
+      };
+
+      await setOutgoingKey();
+      await utils.saveDocs(docs);
+
+      await browser.waitUntil(() => {
+        const requestedBroadcastIds = messagesEndpointRequests.map(([query]) => query.broadcast);
+        return _.uniq(requestedBroadcastIds).length === 7;
+      }, 4000);
+      await browser.pause(1000); // wait for the docs to actually be updated
+
+      const requestedBroadcastIds = [];
+      const expectedBroadcastIds = docs.map(doc => doc.tasks[0].gateway_ref).sort();
+      messagesEndpointRequests.forEach(([ query, headers ]) => {
+        expect(query.broadcast).to.not.be.undefined;
```

Comment:
```suggestion
        expect(query.broadcast).to.exist;
```

This should mean the same thing right?